src/pages/home/ReportScreen.js

```diff
@@ -163,7 +163,8 @@ class ReportScreen extends React.Component {
      * @returns {Boolean}
      */
     shouldShowLoader() {
-        return !getReportID(this.props.route) || (_.isEmpty(this.props.reportActions) && this.props.isLoadingInitialReportActions);
+        const isLoadingInitialReportActions = _.isEmpty(this.props.reportActions) && this.props.isLoadingReportActions;
```

Comment:
@marc suggested this and I agreed, this makes it easier to understand the purpose of the check

Reply:
Can you please update the method description in the JSDocs to be more clear? It is answering "what", but not "why"

Reply:
Shouldn't this really be more like `isLoadingReportActions`? It doesn't look like it has anything to do with "initial" actions or not. I'm not even sure what "initial" implies.

Reply:
"initial" means you have no `reportActions` at all to display vs. having some `reportActions` to display, but still in the process of loading the next set of actions ones. There are two different animations displayed based on which situation we have.

Reply:
are you happy with that @tgolen or do you want me to rename the var?

Reply:
I really like Marc's explanation. Could you add that as a code comment to help explain it to others who will see this? If the comment is added, then I think the name is OK then.

Reply:
Will do

Reply:
I added a shorter version of Marcs answer, that I feel makes more sense to read, let me know if that works.